lib/nats.js

```diff
@@ -843,6 +865,9 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
  */
 
 Client.prototype.subscribe = function(subject, opts, callback) {
+  if (this.closed) {
+    this.emit('error', SUB_CONN_CLOSED_ERR);
```

Comment:
The _client_ object should be persistent through connection losses, though.  It should have its own eventemitter (as it does) to handle events across connections.

Yes, were I to design afresh, with no mind to backward compatibility, I would keep errors in-scope.  I don't think we can safely add an error argument to the callback, though.  (By Node convention, the error parameter is the _first_ parameter to the callback.)

Thus, I don't see any way to adapt this to localize events.  I suppose you _could_ add a second callback, but that starts to get a bit ridiculous.